src/screens/Authentication/Authentication.tsx

```diff
@@ -8,50 +8,45 @@ import { Welcome } from "./Welcome";
 
 interface AuthenticationProps {
   setAccessToken: (token: string) => void;
-  isSetup: boolean;
+  hasProjectId: boolean;
 }
 
 type AuthenticationScreen = "welcome" | "signin" | "subdomain" | "verify";
 
-export const Authentication = ({ setAccessToken, isSetup }: AuthenticationProps) => {
-  const [screen, setScreen] = useState<AuthenticationScreen>(isSetup ? "signin" : "welcome");
+export const Authentication = ({ setAccessToken, hasProjectId }: AuthenticationProps) => {
+  const [screen, setScreen] = useState<AuthenticationScreen>(hasProjectId ? "signin" : "welcome");
 
   const [isMounted, setMounted] = useState(true);
   useEffect(() => () => setMounted(false), []);
-  useEffect(() => {
-    if (isSetup && screen === "welcome") {
-      setScreen("signin");
-    }
-  }, [isSetup, screen]);
 
   const { onSignIn, userCode, verificationUrl } = useSignIn({
     isMounted,
     onSuccess: setAccessToken,
     onFailure: () => {},
   });
 
-  switch (screen) {
-    case "welcome":
+  switch (true) {
```

Comment:
Let's just make these a bunch of `if` statements then. The `switch` serves no purpose now (no fall-through, no real default).